pkg/reconciler/pipelinerun/affinity_assistant.go

```diff
@@ -179,14 +179,6 @@ func (c *Reconciler) cleanupAffinityAssistants(ctx context.Context, pr *v1.Pipel
 			if err := c.KubeClientSet.AppsV1().StatefulSets(pr.Namespace).Delete(ctx, affinityAssistantStsName, metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
 				errs = append(errs, fmt.Errorf("failed to delete StatefulSet %s: %w", affinityAssistantStsName, err))
 			}
-
-			// cleanup PVCs created by Affinity Assistants
-			if w.VolumeClaimTemplate != nil {
-				pvcName := getPersistentVolumeClaimNameWithAffinityAssistant(w.Name, pr.Name, w, *kmeta.NewControllerRef(pr))
-				if err := c.KubeClientSet.CoreV1().PersistentVolumeClaims(pr.Namespace).Delete(ctx, pvcName, metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
-					errs = append(errs, fmt.Errorf("failed to delete PersistentVolumeClaim %s: %w", pvcName, err))
-				}
-			}
```

Comment:
would it make sense to revert only this change?

Reply:
Revert this part only will resulting in pvcs not deleted when the pipelinerun is deleted though

Reply:
And I'm not sure https://github.com/tektoncd/pipeline/pull/6635 can work if only reverting this part 🤔

Reply:
We could add a finalizer for PipelineRuns that deletes any affinity assistant PVCs when the PipelineRun is deleted. Also, can you elaborate on what #6635 might not work?

Reply:
Yeah. If we just revert this part, the pvcs won't be deleted after this PR is merged. But if we can merge #6635 in the same release to delete the `pvcs` when the `pr` is deleted, it should be fine probably.

Reply:
/hold

Reply:
I think we still need to call APIs to delete `pvcs` at `pr` completion/deletion time anyways. 

The advantage of this approach is that there is no extra work to delete the `pvcs` at `pr` deletion time, as the owner of the `pvcs` are the `prs` not the `aa statefulset` (i.e. k8s automatically deletes the pvcs when the owning prs are deleted). 

In addition, this patch fully reverts the pvcs behaviours in 1 pull request. If we just revert the highlighted part but not merge the followup prs in one release, users will still see the different behavior (i.e. `pvcs` are not deleted when `pr` is deleted). 

I personally feel this full revert could be a better solution. I'm open to the partial solution if we can make sure to merge all the followup prs in one pr, but I don't see much advantage.

WDYT @lbernick @pritidesai @afrittoli

Reply:
OK, I think it's a fair point that finalizers don't have much advantage over owner references, and have more unknowns since we haven't tried that approach before. this PR SGTM.

Reply:
Thanks, Lee! 
/hold cancel